### PR TITLE
Totem of Undying functionality and research for Undertaker

### LINKED ISSUE #390: Implement Totem use for Undertaker resurrection
### Prerequisites

- [x] I am running the latest **alpha** version of MineColonies for my Minecraft version.
- [x] I checked the MineColonies/Structurize [wiki](https://wiki.minecolonies.ldtteam.com/) and made sure that my issue is not covered there.
- [x] I made sure that **this change request is not a duplicate of any existing change request.**

## Is your feature request related to a problem?

Improvement discussed over discord for the PR: https://github.com/ldtteam/minecolonies/pull/6737 ( https://github.com/ldtteam/minecolonies/pull/6737#issuecomment-799808444 )

## Enhancement or Change description

Implement the use of Totem by the Undertaker while trying to resurrect a citizen

If the Undertaker has a totem in his inventory: the resurrection chance and cap are increased by+5%,
With 2 totem the resurrection chance and cap are increased by +7.5%
The use of totem by the Undertaker is locked by a research (with the totem symbol)
Providing a totem to the Undertaker will give an achievement
The totem have a 1% break chance on each attempt of resurrection

### Notes or related things

---
## Viewers

* Add a thumbs-up to the change request to vote for it. This helps the request become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

### PR DESCRIPTION
**Pull request by chandler05**
Closes ldtteam/minecolonies-features#390
Closes #
Closes #

# Changes proposed in this pull request:
- Adds an advancement for giving a totem to the undertaker.
- Adds research for undertaker totem usage.
- Gives undertaker ability to use totem of undying to assist in resurrection.

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/undertaker/EntityAIWorkUndertaker.java [hunk 1]
```diff
@@ -326,6 +334,11 @@ private IAIState tryResurrect()
             shouldDumpInventory = true;
             final double chance = getResurrectChance(buildingGraveyard);
 
+            if (getTotemResurrectChance() > 0 && random.nextDouble() <= TOTEM_BREAK_CHANCE) {
+                worker.getInventoryCitizen().extractItem(InventoryUtils.findFirstSlotInItemHandlerWith(worker.getInventoryCitizen(), Items.TOTEM_OF_UNDYING), 1, false);
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/undertaker/EntityAIWorkUndertaker.java:333]: formatting is off

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/undertaker/EntityAIWorkUndertaker.java [hunk 2]
```diff
@@ -121,6 +123,9 @@ private IAIState startWorking()
             getOwnBuilding().ClearCurrentGrave();
         }
 
+        // Check for totem in order to grant advancement
+        getTotemResurrectChance();
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/undertaker/EntityAIWorkUndertaker.java]: Nah, a lot of unnecessary checks, it's fine to just give the advancement only on the attempt to resurrect

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/undertaker/EntityAIWorkUndertaker.java [hunk 3]
```diff
@@ -349,15 +362,42 @@ private IAIState tryResurrect()
...
 
+    /**
+     * Check for a totem of undying, the required research, and get the increased resurrection chance
+     *
+     * @return the chance increase that the totem provides
+     */
+    private double getTotemResurrectChance() {
+        final int totems = InventoryUtils.getItemCountInItemHandler(worker.getInventoryCitizen(), Items.TOTEM_OF_UNDYING);
+
+        if (totems > 0)
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/undertaker/EntityAIWorkUndertaker.java:380]: also formatting is off

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/undertaker/EntityAIWorkUndertaker.java [hunk 4]
```diff
@@ -349,15 +357,45 @@ private IAIState tryResurrect()
...
         if (chance > cap) { chance = cap; }
         return chance;
     }
 
+    /**
+     * Check for a totem of undying, the required research, and get the increased resurrection chance
+     *
+     * @return the chance increase that the totem provides
+     */
+    private double getTotemResurrectChance() {
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/undertaker/EntityAIWorkUndertaker.java]: still wrong formatting

**Review by Mekle001 — APPROVED** [review]

**Review by marchermans — APPROVED** [review]

**Pull request merged**